apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/Span.java

```diff
@@ -41,6 +45,11 @@
     public static final long MAX_LOG_INTERVAL_MICRO_SECS = TimeUnit.MINUTES.toMicros(5);
     private static long lastSpanMaxWarningTimestamp;
 
+    /**
+     * Known sub-types for which reporting an exception will also set the span outcome to 'failure'
+     */
+    private static final Set<String> SUB_TYPES_EXCEPTIONS_OUTCOME_FAILURE = new HashSet<String>(Arrays.asList("dubbo", "redis"));
```

Comment:
here redis is not considered as a "db" span, I'm not sure if it's expected or not.

Reply:
Good point, I'll change that to only rely on type.
For Java, when we have type = "ext" or "external" having any exception would indicate failure.